### PR TITLE
TNO-731: Move work orders

### PR DESCRIPTION
**Pull request by jdtoombs**
Move work orders to the editor tab, and make it read only for non-admins

### CODE REVIEW

File: app/editor/src/store/hooks/admin/useWorkOrders.ts [hunk 1]
```diff
@@ -14,22 +21,24 @@ interface IWorkOrderController {
 }
 
 export const useWorkOrders = (): [IAdminState, IWorkOrderController] => {
-  const api = useApiAdminWorkOrders();
+  const adminApi = useApiAdminWorkOrders();
```
Comment (jdtoombs) [inline app/editor/src/store/hooks/admin/useWorkOrders.ts:24]: Maybe not necessary?

File: app/editor/src/features/admin/work-orders/WorkOrderForm.tsx [hunk 2]
```diff
@@ -83,24 +84,27 @@ export const WorkOrderForm: React.FC = () => {
                   options={userOptions}
                   name="requestorId"
                   label="Requestor"
+                  isDisabled={!isAdmin}
                   value={userOptions.find((s) => s.value === values.requestorId) || ''}
                 />
-                <FormikTextArea name="description" label="Description" />
+                <FormikTextArea name="description" label="Description" disabled={!isAdmin} />
```
Comment (Fosol) [inline app/editor/src/features/admin/work-orders/WorkOrderForm.tsx:90]: Not sure there is any value in allowing non-admins access to the page if they can't do anything.

File: app/editor/src/hooks/useAdmin.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Claim, useKeycloakWrapper } from 'tno-core';
+
+/**
+ * Hook to determine if a user is an admin based off their claims
+ * @returns boolean value indicating if the user is an admin
+ */
+export const useAdmin = () => {
```
Comment (Fosol) [inline app/editor/src/hooks/useAdmin.ts:7]: This hook is a bit overkill.  Unless it has more functionality it is more intuitive for us to just use the `useKeycloakWrapper` hook.

File: app/editor/src/store/hooks/admin/useWorkOrders.ts (hunk 1 above)
Comment (Fosol) [reply app/editor/src/store/hooks/admin/useWorkOrders.ts:24]: I don't think we need `useApiAdminWorkOrders` if we're giving access to editors now.  Move the controller and give appropriate permission.  At some point we'll probably not want editors this kind of access, but for now it'll be okay.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: app/editor/src/hooks/useAdmin.ts (hunk 3 above)
Comment (jdtoombs) [reply app/editor/src/hooks/useAdmin.ts:7]: Sounds good, think I've done this overkill one twice now 😛 getting deja vu

File: app/editor/src/features/admin/work-orders/WorkOrderForm.tsx (hunk 2 above)
Comment (jdtoombs) [reply app/editor/src/features/admin/work-orders/WorkOrderForm.tsx:90]: I think the ticket said allow editors to view, only reason I added that in!

File: libs/net/models/Areas/Editor/Models/Content/WorkOrderModel.cs [hunk 4]
```diff
@@ -70,6 +86,50 @@ public WorkOrderModel(Entities.WorkOrder entity)
         this.AssignedId = entity.AssignedId;
         this.Description = entity.Description;
         this.Note = entity.Note;
+        if (entity.Requestor != null)
+            this.Requestor = new UserModel(entity.Requestor);
+        if (entity.Assigned != null)
+            this.Assigned = new UserModel(entity.Assigned);
+        // if (entity.Content != null)
```
Comment (jdtoombs) [inline libs/net/models/Areas/Editor/Models/Content/WorkOrderModel.cs:93]: Not sure why this is causing an error? Currently headline not being populated as I cannot get this to work

File: api/net/Areas/Editor/Controllers/WorkOrderController.cs [hunk 5]
```diff
@@ -4,7 +4,7 @@
 using Microsoft.AspNetCore.SignalR;
 using Microsoft.Extensions.Options;
 using Swashbuckle.AspNetCore.Annotations;
-using TNO.API.Areas.Editor.Models.WorkOrder;
+using TNO.API.Areas.Editor.Models.Content;
```
Comment (Fosol) [inline api/net/Areas/Editor/Controllers/WorkOrderController.cs:7]: This doesn't align with convention.  Normally we create models for a controller so that editing them doesn't impact another controller.

File: app/editor/src/features/content/form/interfaces/IContentForm.tsx [hunk 6]
```diff
@@ -10,8 +10,8 @@ import {
   IContentTonePoolModel,
   IFileReferenceModel,
   ITimeTrackingModel,
-  IWorkOrderModel,
 } from 'hooks/api-editor';
+import { IWorkOrderModel } from 'store/slices/workorder/interfaces';
```
Comment (Fosol) [inline app/editor/src/features/content/form/interfaces/IContentForm.tsx:14]: I don't understand the need for a different interface.

File: app/editor/src/features/work-orders/utils/makeWorkOrderFilter.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
-import { IWorkOrderFilter } from 'hooks';
+import { IWorkOrderFilter } from 'store/slices/workorder/interfaces';
```
Comment (Fosol) [inline app/editor/src/features/work-orders/utils/makeWorkOrderFilter.ts:1]: I don't understand the need for a different interface.

File: app/editor/src/store/slices/workorder/interfaces/IWorkOrderFilter.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import { WorkOrderStatusName, WorkOrderTypeName } from '../constants';
+import { WorkOrderStatusName, WorkOrderTypeName } from 'hooks';
 
 export interface IWorkOrderFilter {
```
Comment (Fosol) [inline app/editor/src/store/slices/workorder/interfaces/IWorkOrderFilter.ts:3]: Why did you move this?  This interface represents a model in the API.

File: app/editor/src/store/slices/workorder/interfaces/IWorkOrderListFilter.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import { ISortBy } from 'features/content/list-view/interfaces';
+import { WorkOrderStatusName, WorkOrderTypeName } from 'hooks';
+
+export interface IWorkOrderListFilter {
```
Comment (Fosol) [inline app/editor/src/store/slices/workorder/interfaces/IWorkOrderListFilter.ts:4]: This filter seems specific to the WorkOrderList component.  Keep it close to that component the same way we are in the list-view.

File: app/editor/src/store/slices/workorder/interfaces/IWorkOrderModel.ts [hunk 10]
```diff
@@ -4,7 +4,7 @@ import {
   IUserModel,
   WorkOrderStatusName,
   WorkOrderTypeName,
-} from '..';
+} from 'hooks';
 
 export interface IWorkOrderModel extends IAuditColumnsModel {
```
Comment (Fosol) [inline app/editor/src/store/slices/workorder/interfaces/IWorkOrderModel.ts:9]: This is an interface that represents a model from the API.   Why are you moving it here?

File: libs/net/models/Areas/Editor/Models/Content/WorkOrderModel.cs [hunk 11]
```diff
@@ -6,7 +6,7 @@ namespace TNO.API.Areas.Editor.Models.Content;
 /// <summary>
 /// WorkOrderModel class, provides a model that represents a work order.
 /// </summary>
-public class WorkOrderModel
+public class WorkOrderModel : AuditColumnsModel
```
Comment (Fosol) [inline libs/net/models/Areas/Editor/Models/Content/WorkOrderModel.cs:9]: This is incorrect.  Models for the content controller do not need this.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: app/editor/src/features/content/form/interfaces/IContentForm.tsx (hunk 6 above)
Comment (jdtoombs) [reply app/editor/src/features/content/form/interfaces/IContentForm.tsx:14]: A lot of the changes were done as the interfaces were stored in /admin and had been relocated. Did a mass update imports - will look into these
